### DISCUSSION TITLE
Test is getting failed due to this. If anyone faced same issue do let me know the solution.

### DISCUSSION THREAD

**Discussion by ymanish2910** [wevm/wagmi#3327] [Q&A]
import { createSyncStoragePersister } from "@tanstack/query-sync-storage-persister"

SyntaxError: Cannot use import statement outside a module

**Comment by tmm** [comment]
If you are using Jest, you need to set up ESM loading.

**Reply by ymanish2910** [reply]
How to setup ESM loading ?
I attempted by adding "type":"module" in package.json file. But there I found error `ReferenceError: module is not defined in ES module scope
This file is being treated as an ES module because it has a '.js' file extension `.

My Jest config is like
`import nextJest from 'next/jest.js'

const createJestConfig = nextJest({
  dir: './',
})

export default createJestConfig({
  preset: 'ts-jest',
  transform: {
    '^.+\\.(ts|tsx)?$': 'ts-jest',
  },
  testPathIgnorePatterns: ['<rootDir>/.next/', '<rootDir>/node_modules/'],
  testEnvironment: 'jest-environment-jsdom',
  moduleNameMapper: {
    '^@/(.*)$': '<rootDir>/src/$1',
  },
  setupFilesAfterEnv: ['<rootDir>/jest.setup.js'],
})
`

**Reply by tmm** [reply]
I don't use Jest so am not familiar. We use [Vitest](https://vitest.dev/) instead.
